File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for i in s:
            if i=='(' or i=='[' or i=='{':
                stack.append(i)
            else:
                if stack:
                    if i==')' and stack[-1]!='(':
                        return False
                    if i==']' and stack[-1]!='[':
                        return False
                    if i=='}' and stack[-1]!='{':
                        return False
                    stack.pop()
                else:
                    return False
        if stack:
            return False
        else:
            return True

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack <Character> charStack = new Stack<>();
        for (char chars : s.toCharArray()){
            if ((chars == '(') || (chars == '[') || (chars == '{')){
                charStack.push(chars);
            }
            else if ((chars == ')') || (chars == ']') || (chars == '}')){
                if (!charStack.isEmpty())
                {
                    if ((chars == ')') && (charStack.peek() != '(')){
                        return false;
                    }
                    else if ((chars == '}') && (charStack.peek() != '{')){
                        return false;
                    }
                    else if ((chars == ']') && (charStack.peek() != '[')){
                        return false;
                    }
                    charStack.pop();
                }
                else
                {
                    return false;
                }
            }
        }
    if (charStack.isEmpty()){
        return true;
    }
    return false;
    }

}